### ISSUE TITLE
Prevent daily schedule from being executed at 00:00 if the target state does not differ

### ISSUE THREAD

**Issue by LinusHoppe** [nielsfaber/scheduler-component#270]
### Checklist
- [x] I checked for similar existing requests (both open and closed) before posting.
- [x] My request is generic, other users may benefit from it too.

### tl;dr
I need day-overlapping time schemes, e. g. `Monday 22:00 to Tuesday 06:00`. As this is unlikely to happen, I would love to have an option for time schemes to not trigger any action if the previous target state does not differ from the current target state. This would allow me to mark time schemes starting at midnight as "do not trigger".

### Proposal
I have a lot of daily schedules for heating, cover etc. - all of them are configured by hourly ranges (see example). This allows me to change something (target temperature, cover close/open) in between, the target state of the schedule is only executed if the new time slot is reached.
But on midnight 0:00 every day, all daily schedules are triggered even if the target state between the previous day 23:59 and the current day 0:00 does not differ. This results in the fact that all manual actions changes in between are overwritten (e. g. all covers moving to the scheduled position at midnight).

Would it be possible to add a flag to a schedule that on the schedule start time (00:00) no action is triggered if the previous and the current target state does not differ?

### Additional info
![grafik](https://user-images.githubusercontent.com/32911160/236747660-1ec1233a-5976-4c0f-903d-a1803f03ff0e.png)

**Comment by Nightenom** [comment]
I suffer the exactly same issue, ideally no action should be taken at midnight, but at least if this could be removed from ui, so if next change happens tomorrow at 5 am, then it shows 5 am and not midnight

**Comment by Nightenom** [comment]
Still an issue

**Comment by nielsfaber** [comment]
By design the schedules last from 00:00 to 23:59, so if a schedule has an action assigned to the first timeslot (starting at 00:00), there will be an action triggered at midnight.

It is not really possible to change the daily repetition cycle (i.e. creating a single timeslot spanning over multiple days) this would also complicate a lot of things in both the card and the integration.

I also don’t really see a straightforward way how to determine whether a scheduled action ‘has no effect’ and thus a timeslot could be skipped. Since scheduler supports all kind of entities, a lot of logic (per entity type) would have to be implemented to compare the current state with what the state would become after performing a certain action. 

In summary, I agree it would be nice if scheduler could automatically skip timeslot transitions which have no effect, but a large  amount of effort would be needed to implement it and there is not really any functional benefit (other than the next displayed time in the card being more meaningful).

**Comment by Nightenom** [comment]
I completely agree with your reasoning, which brings up a simple question: could you just detect this in ui and adjust the rendering accordingly? Personally I think it just simple value comparison regardless the entity type

**Comment by nielsfaber** [comment]
@Nightenom the calculation of the next timeslot is performed in the integration side. The card only visualizes this info. Hence it is definitely not possible to implement this feature on the UI side.

**Comment by bastiitsab** [comment]
is there any way to prevent this?
for example I could manually open my shutter after sunset. Then the scheduler will close it at 00:00, but it should first do this on sunrise.
